tests/functional/api/test_merge_requests.py

```diff
@@ -183,6 +183,7 @@ def test_merge_request_reset_approvals(gitlab_url, project, wait_for_sidekiq):
 
 def test_cancel_merge_when_pipeline_succeeds(project, merge_request, wait_for_sidekiq):
     mr = merge_request(source_branch="test_merge_request_merge", create_pipeline=True)
+    wait_for_sidekiq(timeout=60)
```

Comment:
   Not sure I know what the sidekiq so maybe a dumb question. Does it check that the merge requested created, the pipeline triggered and finished running?

Reply:
I'm not an expert but I believe sidekiq is what takes care of handling the tasks that get run asynchronously. So by waiting we are hoping that all pending tasks will be done before we continue.

Reply:
Yes the original idea was to wait for all pending/asynchronous jobs to complete, with the assumption the server is done with whatever we want it to :) probably not the best way to do it, but kind of better than the random  hardcoded guesswork with `time.sleep()` we had before.